tests/serialized_test_data_generation/serialbox_to_netcdf.py

```diff
@@ -0,0 +1,134 @@
+    )
+    return parser
+
+
+def read_serialized_data(serializer, savepoint, variable):
+    data = serializer.read(variable, savepoint)
+    if len(data.flatten()) == 1:
+        return data[0]
+    return data
+
+
+def get_all_savepoint_names(data_path):
+    savepoint_names = set()
+    serializer = get_serializer(data_path, rank=0)
+    for savepoint in serializer.savepoint_list():
+        savepoint_names.add(savepoint.name)
+    return savepoint_names
+
+
+def get_serializer(data_path, rank):
+    return serialbox.Serializer(
+        serialbox.OpenModeKind.Read, data_path, "Generator_rank" + str(rank)
+    )
+
+
+def main(data_path: str, output_path: str):
+    os.makedirs(output_path, exist_ok=True)
+    namelist_filename_in = os.path.join(data_path, "input.nml")
+    namelist_filename_out = os.path.join(output_path, "input.nml")
+    if namelist_filename_out != namelist_filename_in:
+        shutil.copyfile(os.path.join(data_path, "input.nml"), namelist_filename_out)
+    namelist = f90nml.read(namelist_filename_out)
+    total_ranks = 6 * namelist["fv_core_nml"]["layout"][0] * namelist["fv_core_nml"]["layout"][1]
+    grid_list = []
+    for rank in range(total_ranks):
+        serializer = get_serializer(data_path, rank)
+        grid_savepoint = serializer.get_savepoint("Grid-Info")[0]
+        grid_data = {}
+        grid_fields = serializer.fields_at_savepoint(grid_savepoint)
+        for field in grid_fields:
+            grid_data[field] = read_serialized_data(serializer, grid_savepoint, field)
+        grid_list.append(grid_data)
+    data_vars = {}
+    for varname in grid_list[0].keys():
+        serialbox_example = grid_list[0][varname]
+        rank_data = get_grid_data(serialbox_example, total_ranks, grid_list, varname)
+        data_vars[varname] = rank_data
+    dataset = xr.Dataset(data_vars=data_vars)
+    dataset.to_netcdf(os.path.join(output_path, f"Grid-Info.nc"))
+
+    savepoint_names = get_all_savepoint_names(data_path)
+    for savepoint_name in sorted(list(savepoint_names)):
+        rank_list = []
+        names_list = []
+        serializer_list = []
+        for rank in range(total_ranks):
+            serializer = get_serializer(data_path, rank)
+            serializer_list.append(serializer)
+            savepoints = serializer.get_savepoint(savepoint_name)
+            if rank == 0:
+                names_list.extend(serializer.fields_at_savepoint(savepoints[0]))
+            rank_data = {}
+            for name in set(names_list):
+                rank_data[name] = []
+                for savepoint in savepoints:
+                    rank_data[name].append(read_serialized_data(serializer, savepoint, name))
+            rank_list.append(rank_data)
+        n_savepoints = len(savepoints)  # checking from last rank is fine
+        data_vars = {}
+        if n_savepoints > 0:
+            for varname in set(names_list).difference(["rank"]):
+                serialbox_example = rank_list[0][varname][0]
```

Comment:
Is this because we're just using it as example to get the shape? Can we just pass the shape instead?